packages/client/src/containers/CollectionsPage/CollectionsPage.Container.js

```diff
@@ -1,12 +1,91 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import ProductList from '../../components/ProductList/ProductList.component';
-import DropDownViewList from '../../components/ProductListDropDown/DropDownViewList/DropDownViewList.component';
+import './CollectionsPage.style.css';
+import DropDownView from '../../components/CategoriesListDropDown/CategoriesListDropDown.component';
+import { apiURL } from '../../apiURL';
 
 export const CollectionsPage = () => {
+  const [products, setProducts] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [onSelectedCategory, setOnSelectedCategory] = useState('');
+  const [onSelectedSortBy, setOnSelectedSortBy] = useState('');
+  // const [onSelectedAllFilter, setOnSelectedAllFilter] = useState('');
+
+  const fetchProducts = () => {
+    setIsLoading(true);
+    fetch(`${apiURL()}/products/`)
+      .then((res) => res.json())
+      .then((item) => {
+        setProducts(item);
+      })
+      .then(() => {
+        setIsLoading(false);
+      });
+  };
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  const [category, setCategoy] = useState([]);
+  const fetchCategories = () => {
+    fetch(`${apiURL()}/categories/`)
+      .then((res) => res.json())
+      .then((item) => {
+        setCategoy(item);
+      });
+  };
+  useEffect(() => {
+    fetchCategories();
+  }, []);
+
+  const categories = category.map(
+    (element) =>
+      element.name.charAt(0).toUpperCase() +
+      element.name.slice(1).toLowerCase(),
+  );
+
+  const sortBy = ['Recent Collections', 'Alphabetically', 'Price ↓', 'Price ↑'];
+  const allFilter = ['Price', 'Size', 'Color', 'Reviews', 'Brand'];
+
+  const fetchDropdown = (categoryId) => {
+    fetch(`${apiURL()}/products?category=${categoryId}`)
+      .then((res) => res.json())
+      .then((item) => {
+        setProducts(item);
+      });
+  };
+  useEffect(() => {
+    fetchDropdown(onSelectedCategory);
+  }, [onSelectedCategory]);
```

Comment:
This part still confuses me, we fetch products twice in different places.

I would instead merge this into one place but check if you can add the category filter if it's set, something like:
```
useEffect(() => {
  function fetchProducts() {
    // Still, how do you get the category ID here?
    const url = `${apiURL()}/products${selectedCategory ? '?category=${categoryId}' : ''}`;
  
    fetch(url)
      .then((res) => res.json())
      .then((items) => {
        setProducts(items);
      });
  };

  fetchProducts();
}, [selectedCategory]);
```

Reply:
Its my mistake. Its fetch category Name not categoryId  but Yea, I will use useEffect for render product by category name.

Reply:
I have merge product and query param in one fetch.  I have passed category name as fetch function parameter. 
useEffect(() => {
    function fetchProducts(categoryName) {
      const url = `${apiURL()}/products${selectedCategory ? `?category=${categoryName}` : ''}`;
      fetch(url)
        .then((res) => res.json())
        .then((items) => {
          setProducts(items);
        });
    }
    fetchProducts(selectedCategory);
  }, [selectedCategory]);

Reply:
Yes.. that true.. but we create endpoint by name.